### PR TITLE
Reverse bridging

### PR DESCRIPTION
**Pull request by yahgwai**
Introduces a new custom gateway that allows users to mint tokens and track supply on the L2 rather than L1. Tokens on L2 are escrowed by the gateway, then bridgeMint/burned on L1

### CODE REVIEW

File: contracts/tokenbridge/ethereum/gateway/L1ReverseCustomGateway.sol [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * @title   L1 Gateway for reverse "custom" bridging functionality
+ * @notice  Handles some (but not all!) reverse custom Gateway needs.
+ *          Use the reverse custom gateway instead of the normal custom
+ *          gateway if you want total supply to be tracked on the L2
+ *          rather than the L1.
+ * @dev     The reverse custom gateway burns on the l2 and escrows on the l1
+ *          which is the opposite of the way the normal custom gateway works
+ *          This means that the total supply L2 isn't affected by briding, which
```
Comment (DZGoldman) [inline contracts/tokenbridge/ethereum/gateway/L1ReverseCustomGateway.sol:34]:
```suggestion
 *          This means that the total supply L2 isn't affected by bridging, which
```

File: contracts/tokenbridge/ethereum/gateway/L1ReverseCustomGateway.sol [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ *          is helpful if the minting is also occuring on L2
```
Comment (DZGoldman) [inline contracts/tokenbridge/ethereum/gateway/L1ReverseCustomGateway.sol:35]: >  is helpful if the minting is also occuring [sic] on L2

I would phrase it more like its "helpful if you want the L2 to be able to directly calculate the circulating supply (IOW, L2 minting we can support with the non-reverse custom gateway)

File: contracts/tokenbridge/ethereum/gateway/L1ArbitrumGateway.sol [hunk 3]
```diff
@@ -24,7 +24,6 @@ import "@openzeppelin/contracts/utils/Create2.sol";
 import "@openzeppelin/contracts/utils/Address.sol";
 import "@openzeppelin/contracts/math/SafeMath.sol";
 
-
```
Comment (DZGoldman) [inline contracts/tokenbridge/ethereum/gateway/L1ArbitrumGateway.sol:27]: 86 it

**Review by DZGoldman — APPROVED** [review]

File: contracts/tokenbridge/ethereum/gateway/L1ReverseCustomGateway.sol [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ *          This means that the total supply L2 isn't affected by bridging, which
+ *          is helpful if the minting is also occuring on L2
+ */
+contract L1ReverseCustomGateway is L1CustomGateway {
+    function inboundEscrowTransfer(
+        address _l1Address,
+        address _dest,
+        uint256 _amount
+    ) internal virtual override {
+        IArbToken(_l1Address).bridgeMint(_dest, _amount);
```
Comment (gzeoneth) [inline contracts/tokenbridge/ethereum/gateway/L1ReverseCustomGateway.sol:43]: risk of compromised bridge owner being able to mint infinite supply of all token registered to the new `L1ReverseCustomGateway` by force registering malicious pair

**Pull request merged**